tests/mock_network/mod.rs

```diff
@@ -572,13 +572,37 @@ fn carry_out_parsec_pruning() {
 }
 
 #[test]
-fn node_pause_and_resume() {
+fn node_pause_and_resume_simple() {
     let network = Network::new(NetworkParams {
         elder_size: LOWERED_ELDER_SIZE,
         safe_section_size: LOWERED_ELDER_SIZE,
     });
-    let mut nodes = create_connected_nodes(&network, 8);
+    let nodes = create_connected_nodes(&network, 2 * network.safe_section_size() - 2);
+    let new_node_id = FullId::gen(&mut network.new_rng());
+    node_pause_and_resume(network, nodes, new_node_id)
+}
+
+#[test]
+fn node_pause_and_resume_during_split() {
+    let network = Network::new(NetworkParams {
+        // The paused node does not participate until resumed, so we need enough elders to reach
+        // consensus even without it.
+        elder_size: 4,
+        safe_section_size: 4,
+    });
+
+    let mut nodes = create_connected_nodes(&network, network.safe_section_size());
+    let prefix =
+        add_connected_nodes_until_one_away_from_split(&network, &mut nodes, &[Prefix::default()])
+            [0];
 
+    let new_node_id = FullId::within_range(&mut network.new_rng(), &prefix.range_inclusive());
+    node_pause_and_resume(network, nodes, new_node_id)
+}
+
+// Pause a random node, then add new node with the given id, then resume the paused node and verify
+// everything still works as expected.
```

Comment:
so the expected behaviour of this resumed node is that it will be given a new id?

Reply:
No. What the test does is it pauses a random node, then adds new node (different one), then resumes the paused node. So the `new_node_id` argument is for the newly added node, not the paused node.

Reply:
ah, ok. missed that